airqo_monitor/format_data.py

```diff
@@ -13,81 +14,98 @@ def parse_field8_metadata(field8):
     """
     Parses thingspeak json response.
-    Field mapping:
-    field1: pm1
-    field2: pm2.5
-    field3: pm10
-    field4: sample period
-    field5: latitude
-    field6: longtitude
-    field7: battery voltage
-    field8 (optional): lat,lng,elevation,speed,num_satellites,hdop
+    Field mapping depends on the channel type's data format
     """
+    channel_id = channel.channel_id
+    channel_type_name = channel.channel_type.name
+    data_format = channel.channel_type.data_format
+
     data = get_data_for_channel(channel_id, start_time=start_time, end_time=end_time)
-    entry_objects = []
+    formatted_data = []
+
     for entry in data:
-        entry_object = DataEntry(
-            channel_id=channel_id,
-            entry_id=entry['entry_id'],
-        )
-        entry_object.created_at = entry['created_at']
-        entry_object.pm_1 = entry['field1']
-        entry_object.pm_2_5 = entry['field2']
-        entry_object.pm_10 = entry['field3']
-        entry_object.sample_period = entry['field4']
-        entry_object.battery_voltage = entry['field7']
-
-        field8 = entry.get('field8', None)
-        if field8:
-            try:
-                lat, lng, altitude, speed, num_satellites, hdop = parse_field8_metadata(field8)
-                entry_object.latitude = lat
-                entry_object.longitude = lng
-                entry_object.altitude = altitude
-                entry_object.speed = speed
-                entry_object.num_satellites = num_satellites
-                entry_object.hdop = hdop
-            except ValueError:
-                # there must have been a misformatted field, fallback to the old
-                # lat and lng fields
-                entry_object.latitude = entry['field5']
-                entry_object.longitude = entry['field6']
-        else:
-            entry_object.latitude = entry['field5']
-            entry_object.longitude = entry['field6']
-
-        # we currently only have stationary devices active
-        entry_object.is_mobile = False
-
-        entry_objects.append(entry_object)
-    return entry_objects
-
-
-def _update_db_channel_table(channel_ids_to_names):
-    for channel_id, channel_info in channel_ids_to_names.items():
+        entry_data = dict()
+        for key, value in data_format.items():
+            entry_data[value] = entry.get(key, None)
+        entry_data['type'] = channel_type_name
+        entry_data['entry_id'] = entry['entry_id']
+        entry_data['channel_id'] = channel_id
+        entry_data['created_at'] = entry['created_at']
+        formatted_data.append(entry_data)
+
+    return formatted_data
+
+
+def update_all_channels_for_channel_type(channel_type):
+    """
+    Given a channel type, get all channels for this channel type from Thingspeak
+    and update them in the DB
+    """
+    all_channels = get_all_channels_cached()
+    channels_for_type = []
+    for channel_data in all_channels:
```

Comment:
how about:

matching_channels = [channel for channel in all_channels if channel_type.name in channel['tags']]

then for channel_data in matching_channels:
    ...